src/bootstrap/configure.py

```diff
@@ -494,11 +494,35 @@ def configure_top_level_key(lines, top_level_key, value):
     for section in section_order:
         if section == 'target':
             for target in targets:
+                # only blocks with an uncommented line will be output
+                block = []
+                has_non_comment = False
                 for line in targets[target]:
-                    f.write(line + "\n")
+                    block.append(line)
+                    if len(line) == 0:
+                        if has_non_comment:
+                            for l in block:
+                                f.write(l + "\n")
+                        block = []
+                        has_non_comment = False
+                        continue
+                    if not line.startswith('#'):
+                        has_non_comment = True
         else:
+            # only blocks with an uncommented line will be output
+            block = []
+            has_non_comment = False
```

Comment:
comment check can be simplified with:

```suggestion
                    block.append(line)
                    if len(line) == 0:
                        if not is_comment:
                            for l in block:
                                f.write(l + "\n")
                        block = []
                        is_comment = True
                        continue
                    is_comment = line.startswith('#'):
        else:
            # only blocks with an uncommented line will be output
            block = []
            is_comment = True
```

Reply:
I don't think this will work if the comment is the last line in the block e.g.:```
# some text
some_config=true
# some more text
end of block```

Reply:
I haven't tested it. Just tried to reverse the idea of `has_non_comment` with `is_comment` and simplify the following block

```python
  if not line.startswith('#'):
      has_non_comment = True
```

with

```python
is_comment = line.startswith('#'):
```

to make it easier to read.

Maybe I mischanged something, is it not possible to write it that way?